src/Compilers/Core/Portable/InternalUtilities/SpecializedCollections.Empty.Set.cs

```diff
@@ -10,8 +10,11 @@ internal partial class SpecializedCollections
         private partial class Empty
         {
             internal class Set<T> : Collection<T>, ISet<T>
+#if COMPILERCORE
```

Comment:
https://github.com/dotnet/roslyn/pull/12353#discussion-diff-74468564

I realize now that I don't need to wait for this to be merged to fix the issue. I can submit a PR ASAP to closed/ to pull in IReadOnlySet.cs, then remove these. Will do!

I fully agree about ifdefs!